jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -1362,30 +1362,33 @@ ecma_regexp_exec_helper (ecma_value_t regexp_value, /**< RegExp object */
     }
   }
 
-  if (input_curr_p && (re_ctx.flags & RE_FLAG_GLOBAL))
+  if (!ECMA_IS_VALUE_ERROR (ret_value))
```

Comment:
What if we merge the two `if`-s? Is it doable?
```c
if (!ECMA_IS_VALUE_ERROR (ret_value) && input_curr_p && (re_ctx.flags & RE_FLAG_GLOBAL)) {...
```

Reply:
Good idea!